app/org/sagebionetworks/bridge/validators/HealthDataRecordValidator.java

```diff
@@ -106,6 +106,12 @@ public void validate(Object object, Errors errors) {
                 errors.rejectValue("uploadDate", Validate.CANNOT_BE_NULL);
             }
 
+            // userMetadata is optional, but if specified, it must be an object node.
+            JsonNode userMetadata = record.getUserMetadata();
+            if (userMetadata != null && !userMetadata.isObject()) {
+                errors.rejectValue("userMetadata", "must be an object node");
+            }
```

Comment:
So, do we ever validate that the JSON expresses the FieldDefinitions defined for a study? Shouldn't we? At least the keys of the object should match.

Reply:
No. As it turns out, too much validation makes things too rigid and slows down app development. It's the same reason why most schema fields end up being marked required=false and why StrictValidation, in hindsight, was a mistake.